### PR TITLE
Alpine build

### PR DESCRIPTION
**Pull request by r1viollet**
# What does this PR do?

Adapt the build process to work on Alpine
This includes a new docker image to pull down the appropriate dependencies for alpine

co-authored with @sanchda 

# Motivation

Compatibility with alpine
A [ticket](https://github.com/DataDog/ddprof/issues/88) was opened requesting this

# Additional Notes

The CI was also adapted to test alpine.
One unit test is still causing issues. savecontext-ut
I had to downgrade it slightly for now. We suspect this is related to the usage of a signal handler.

# How to test the change?

Tests pass on alpine.

### CODE REVIEW

File: tools/fetch_elfutils.sh [hunk 1]
```diff
@@ -68,6 +68,18 @@ patch -p1 < "${SCRIPT_DIR}/elfutils.patch"
 if [[ "$(basename "${C_COMPILER}")" == clang* ]]; then
   export CFLAGS="-Wno-xor-used-as-pow -Wno-gnu-variable-sized-type-not-at-end -Wno-unused-but-set-parameter"
 fi
+
+# Detect musl
+# If so, then set the DFNM_EXTMATCH macro to a harmless value.  We don't use the affected
+# codepaths, so it's OK (TODO: can we just disable compilation of those units?)
+MUSL_LIBC=$(ldd /bin/ls | grep 'musl' | head -1 | cut -d ' ' -f1 || true)
```
Comment (nsavoire) [inline tools/fetch_elfutils.sh:75]: You could have taken build recipe / patches from: 
https://gitlab.alpinelinux.org/alpine/aports/-/blob/master/main/elfutils/APKBUILD

Comment (r1viollet) [reply tools/fetch_elfutils.sh:75]: Fixed, thanks

File: tools/fetch_elfutils.sh [hunk 2]
```diff
@@ -68,6 +69,23 @@ patch -p1 < "${SCRIPT_DIR}/elfutils.patch"
...
+if [ ! -z ${MUSL_LIBC-""} ]; then
+  CFLAGS="${CFLAGS-""} -DFNM_EXTMATCH=0"
```
Comment (nsavoire) [inline tools/fetch_elfutils.sh:78]: I don't think this is needed aymore

Comment (r1viollet) [reply tools/fetch_elfutils.sh]: Fixed, I'll update the build caches

**Review by nsavoire — APPROVED** [review]
Thanks for all the work !
LGTM

**Pull request merged**